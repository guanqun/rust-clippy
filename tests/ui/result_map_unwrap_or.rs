#![warn(clippy::result_map_unwrap_or)]

fn main() {
    let res: Result<usize, ()> = Ok(1);

    // Check `RESULT_MAP_OR_NONE`.
    // Single line case.
    let _ = res.map(|x| x + 1).unwrap_or(0);
    // Multi-line case.
    #[rustfmt::skip]
    let _ = res.map(|x| {
                        x + 1
                       }).unwrap_or(0);
}
