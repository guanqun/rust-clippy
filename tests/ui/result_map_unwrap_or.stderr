error: called `map(f).unwrap_or(a)` on a Result value. This can be done more directly by calling `map_or(a, f)` instead
  --> $DIR/result_map_unwrap_or.rs:8:13
   |
LL |     let _ = res.map(|x| x + 1).unwrap_or(0);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `-D clippy::result-map-unwrap-or` implied by `-D warnings`
   = note: replace `map(|x| x + 1).unwrap_or(0)` with `map_or(0, |x| x + 1)`

error: called `map(f).unwrap_or(a)` on a Result value. This can be done more directly by calling `map_or(a, f)` instead
  --> $DIR/result_map_unwrap_or.rs:11:13
   |
LL |       let _ = res.map(|x| {
   |  _____________^
LL | |                         x + 1
LL | |                        }).unwrap_or(0);
   | |______________________________________^

error: aborting due to 2 previous errors

